src/recipe.c

```diff
@@ -545,6 +546,12 @@ recipe_handler (gpointer user_data)
 
     switch (app_data->state) {
         case RECIPE_FETCH:
+            if (!app_data->stdin
+                && BKR_ENV_EXISTS ()
+                && recipe_wait_on_beaker (app_data->recipe_url, "* Recipe fetch")) {
```

Comment:
I was thinking to put all 3 conditionals within recipe_wait_on_beaker().  This includes the stdin check and BKR_ENV_EXISTS().